specification/http/1.0/openapi.json

```
@@ -0,0 +1,456 @@
+                    "$schema": {
+                      "type": "string",
+                      "format": "uri"
+                    },
+                    "response": {
+                      "$ref": "#/components/schemas/ExecuteToolResponse"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "components": {
+      "securitySchemes": {
+        "bearerAuth": {
+          "type": "http",
+          "scheme": "bearer",
+          "bearerFormat": "JWT"
+        }
+      },
+      "schemas": {
+        "ToolId": {
+          "type": "string",
+          "description": "A tool's unique identifier in the format 'Toolkit.Tool[@version]', where @version is optional.",
+          "pattern": "^[A-Za-z0-9_]+\\.[A-Za-z0-9_]+(@[A-Za-z0-9_.\\-]+)?$",
+          "examples": [
+            "MyToolkit.MyTool",
+            "MyToolkit.MyTool@1.0",
+            "Calculator.Add@latest"
+          ]
+        },
+        "ToolSecret": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string",
+              "description": "The secret's unique identifier."
+            },
+            "value": {
+              "type": "string",
+              "description": "The secret's value."
+            }
+          },
+          "required": ["id", "value"],
+          "additionalProperties": false
+        },
+        "SimpleJSONSchema": {
+          "type": "object",
+          "properties": {
+            "type": "object",
+            "additionalProperties": true,
+            "not": {
+              "properties": {
+                "$ref": {},
+                "$defs": {},
+                "definitions": {}
+              }
+            }
+          }
+        },
+        "ToolDefinition": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "$ref": "#/components/schemas/ToolId"
+            },
+            "name": {
+              "type": "string",
+              "description": "The tool's name."
+            },
+            "description": {
+              "type": "string",
+              "description": "A human-readable explanation of the tool's purpose. This field can be used by both humans and AI models."
+            },
+            "toolkit": {
+              "type": "object",
+              "properties": {
+                "name": {
+                  "type": "string",
+                  "description": "The name of the toolkit."
+                },
+                "description": {
+                  "type": "string",
+                  "description": "A human-readable description of the toolkit."
+                },
+                "version": {
+                  "type": "string",
+                  "description": "An identifier for this version of the toolkit."
+                }
+              },
+              "required": ["name", "description"],
```

id is required during discovery if we use it for tool execution

```suggestion
              "required": ["id", "name", "description"],
```